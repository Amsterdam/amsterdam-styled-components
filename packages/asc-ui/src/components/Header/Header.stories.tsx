import * as React from 'react'
import { storiesOf } from '@storybook/react'
import { action } from '@storybook/addon-actions'
import styled from '@datapunt/asc-core'
import { ReactComponent as ChevronRight } from '@datapunt/asc-assets/lib/Icons/ChevronRight.svg'
import { ReactComponent as ChevronDown } from '@datapunt/asc-assets/lib/Icons/ChevronDown.svg'
import { ReactComponent as ExternalLink } from '@datapunt/asc-assets/lib/Icons/ExternalLink.svg'
import { ReactComponent as MenuIcon } from '@datapunt/asc-assets/lib/Icons/Menu.svg'
import BoxWrapper from '../../internals/Box/BoxWrapper'
import Header, {
  HeaderWrapperStyle,
  SearchBarStyle,
  MenuBarStyle,
  MenuDropDownStyle,
} from './Header'
import ContentFiller from '../../internals/ContentFiller/ContentFiller'
import Typography from '../Typography'
import SearchBar from '../SearchBar'
import Menu from '../Menu/Menu'
import { MenuItem, MenuLabel } from '../..'
import SubMenu from '../Menu/SubMenu'
import { breakpoint } from '../../utils'

const outsideBackgoundColor = '#E6E6E6'
const contentBackgrountColor = '#ffffff'

const HeaderShortStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall={false}
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth
    />
    <ContentFiller backgroundColor={contentBackgrountColor} />
  </BoxWrapper>
)

const HeaderShortContentStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall={false}
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    />
    <ContentFiller backgroundColor={contentBackgrountColor} maxWidth="1800px" />
  </BoxWrapper>
)

const HeaderTallStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    />
    <ContentFiller
      backgroundColor={contentBackgrountColor}
      maxWidth="1800px"
      headerSize="tall"
    />
  </BoxWrapper>
)

const HeaderTallWithContentStory: React.FC<{}> = () => (
  <BoxWrapper backgroundColor={outsideBackgoundColor}>
    <Header
      tall
      title="Data en informatie"
      homeLink="http://data.amsterdam.nl"
      fullWidth={false}
    >
      <Typography element="p">
        Lorem ipsum dolor sit amet, consectetur adipisicing elit. Amet debitis,
        dicta incidunt maxime necessitatibus voluptatibus.
      </Typography>
    </Header>
    <ContentFiller
      backgroundColor={contentBackgrountColor}
      maxWidth="1800px"
      headerSize="tall"
    />
  </BoxWrapper>
)

const DataportaalHeaderWrapperStyle = styled(HeaderWrapperStyle)`
  ${SearchBarStyle} {
    flex-grow: 1;
    & > div {
      flex-grow: 0;
      width: 100%;
      max-width: 430px;

      @media screen and ${breakpoint('max-width', 'mobileM')} {
        max-width: 245px;
      }
    }
  }

  ${MenuBarStyle} {
    @media screen and ${breakpoint('max-width', 'tablet')} {
      display: none;
    }
  }

  ${MenuDropDownStyle} {
    @media screen and ${breakpoint('min-width', 'tablet')} {
      display: none;
    }
  }
`

const MenuDesktop = (
  <Menu>
    <MenuItem onClick={action('click')}>One</MenuItem>
    <SubMenu label="Two">
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuLabel>Two</MenuLabel>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuItem icon={<ExternalLink />} onClick={action('click')}>
        Show more
      </MenuItem>
    </SubMenu>
    <MenuItem onClick={action('click')}>Three</MenuItem>
  </Menu>
)

const MenuMobile = (
  <Menu icon={<MenuIcon />} mobile>
    <MenuItem onClick={action('click')}>One</MenuItem>
    <SubMenu arrowIcon={<ChevronDown />} label="Two">
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuLabel>Two</MenuLabel>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        One
      </MenuItem>
      <MenuItem icon={<ChevronRight />} onClick={action('click')}>
        Two
      </MenuItem>
      <MenuItem icon={<ExternalLink />} onClick={action('click')}>
        Show more
      </MenuItem>
    </SubMenu>
    <MenuItem onClick={action('click')}>Three</MenuItem>
  </Menu>
)

const SearchBarDesktop: React.FC<{}> = () => {
  const [searchText, setText] = React.useState('')
  return (
    <SearchBar
      placeholder="Enter the search text"
      onTextChanged={(text: string) => {
        setText(text)
        action(`text changed: ${searchText}`)
      }}
      onSearch={() => {
        action(`button clicked: ${searchText}`)
      }}
      text={searchText}
    />
  )
}

// const SearchBarMobile = (
//   <SearchBar
//     placeholder="Enter the search text"
//     onTextChanged={(text: string) => {
//       setText(text)
//       action(`text changed: ${searchText}`)
//     }}
//     onSearch={() => {
//       action(`button clicked: ${searchText}`)
//     }}
//     text={searchText}
//   />
// )

const HeaderTallDataportaalStory: React.FC<{}> = () => {
  return (
    <BoxWrapper backgroundColor={outsideBackgoundColor}>
      <Header
        tall
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        fullWidth={false}
        styledComponent={DataportaalHeaderWrapperStyle}
      >
        <SearchBarDesktop />
        {MenuDesktop}
        {MenuMobile}
      </Header>
      <ContentFiller
        backgroundColor={contentBackgrountColor}
        maxWidth="1800px"
        headerSize="tall"
      />
    </BoxWrapper>
  )
}

storiesOf('Composed/Header', module)
  .add('Header short', () => <HeaderShortContentStory />)
  .add('Header short, full width', () => <HeaderShortStory />)
  .add('Header tall', () => <HeaderTallStory />)
  .add('Header tall with header content', () => <HeaderTallWithContentStory />)
  .add('Header tall for dataportaal', () => <HeaderTallDataportaalStory />)
