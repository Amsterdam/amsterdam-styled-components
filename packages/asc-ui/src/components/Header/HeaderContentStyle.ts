import styled from '@datapunt/asc-core'
import { breakpoint } from '../../utils'

const HeaderContentStyle = styled.div`
  height: 50px;
  margin: 0;
  padding: 15px 25px 0 0;
  font-weight: 700;
  width: 70%;
  display: flex;
  justify-content: normal;
  align-self: flex-start;
  justify-content: flex-end;
  align-items: center;
  flex-shrink: 0;
  box-sizing: border-box;

  @media screen and ${breakpoint('max-width', 'laptopM')} {
    display: none;
  }
`
export default HeaderContentStyle
