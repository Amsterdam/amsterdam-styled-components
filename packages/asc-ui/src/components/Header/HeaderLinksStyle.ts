import styled from '@datapunt/asc-core'
import { color, breakpoint, svgFill } from '../../utils'
import { MenuItemStyle } from '../Menu/MenuItem'
import { MenuItemLabelStyle } from '../Menu/MenuItemLabel'

const HeaderLinksStyle = styled.div`
  display: flex;
  position: absolute;
  top: 17px;
  right: 45px;
  align-items: center;
  justify-content: flex-end;

  ${MenuItemStyle} {
    height: 17px;
    padding: 0;

    &:not(:last-child) {
      margin-right: 30px;
    }

    &:focus,
    &:hover {
      background-color: transparent;

      ${MenuItemLabelStyle} {
        color: ${color('secondary')};
      }

      ${svgFill('secondary')};
    }

    ${MenuItemLabelStyle} {
      line-height: 17px;
    }
  }

  @media screen and ${breakpoint('max-width', 'laptopM')} {
    display: none;
  }
`

export default HeaderLinksStyle
