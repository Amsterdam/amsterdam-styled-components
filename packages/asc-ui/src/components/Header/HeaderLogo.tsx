import React from 'react'
import { ReactComponent as Logo } from '@datapunt/asc-assets/lib/Icons/Andreas.svg'
import HeaderLogoStyle, {
  HeaderLogoSetStyle,
  HeaderLogoTextStyle,
} from '../../styles/components/HeaderStyle/HeaderLogoStyle'

type Props = {
  homeLink?: string
}

const HeaderLogo: React.FC<Props> = ({ homeLink }) => (
  <HeaderLogoStyle href={homeLink}>
    <HeaderLogoSetStyle>
      <Logo />
    </HeaderLogoSetStyle>
    <HeaderLogoTextStyle>
      <span>Gemeente</span>
      <span>Amsterdam</span>
    </HeaderLogoTextStyle>
  </HeaderLogoStyle>
)

export default HeaderLogo
