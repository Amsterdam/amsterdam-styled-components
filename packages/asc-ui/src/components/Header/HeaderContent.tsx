import { FunctionComponent, HTMLAttributes, ReactNode } from 'react'
import HeaderContentStyle from './HeaderContentStyle'

type HeaderContentProps = {
  children: ReactNode
} & HTMLAttributes<HTMLDivElement>

const HeaderContent: FunctionComponent<HeaderContentProps> = ({
  children,
  ...otherProps
}) => <HeaderContentStyle {...otherProps}>{children}</HeaderContentStyle>

export default HeaderContent
