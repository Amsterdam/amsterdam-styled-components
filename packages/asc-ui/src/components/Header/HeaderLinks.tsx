import React, { FunctionComponent, HTMLAttributes } from 'react'
import HeaderLinksStyle from './HeaderLinksStyle'

const HeaderLinks: FunctionComponent<HTMLAttributes<HTMLDivElement>> = ({
  children,
  ...otherProps
}) => <HeaderLinksStyle {...otherProps}>{children}</HeaderLinksStyle>

export default HeaderLinks
