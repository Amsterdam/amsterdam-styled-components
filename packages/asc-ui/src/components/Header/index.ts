import HeaderStyle from './HeaderStyle'
import HeaderLinksStyle from './HeaderLinksStyle'
import HeaderTitleStyle from './HeaderTitleStyle'
import HeaderLogoTextStyle from './HeaderLogoTextStyle'
import HeaderNavigationStyle from './HeaderNavigationStyle'

export { default } from './Header'

export const HeaderStyles = {
  HeaderLinksStyle,
  HeaderLogoTextStyle,
  HeaderNavigationStyle,
  HeaderTitleStyle,
  HeaderStyle,
}
