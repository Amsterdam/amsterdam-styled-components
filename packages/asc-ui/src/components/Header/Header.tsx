import React from 'react'
import HeaderStyle, { Props as HeaderStyleProps } from './HeaderStyle'
import HeaderWrapperStyle, {
  Props as HeaderWrapperProps,
} from './HeaderWrapperStyle'
import HeaderLogoText, { LogoProps } from './HeaderLogoText'
import HeaderNavigation from './HeaderNavigation'
import HeaderLinks from './HeaderLinks'
import { CustomCssPropsType } from '../../utils'
import AmsterdamLogo from '../AmsterdamLogo'

type Props = {
  tall?: boolean
  homeLink: string
  title?: string
  navigation?: React.ReactNode
  links?: React.ReactNode
  logo?: React.FC<LogoProps>
} & HeaderWrapperProps &
  HeaderStyleProps &
  CustomCssPropsType &
  React.HTMLAttributes<HTMLDivElement>

const Header: React.FC<Props> = ({
  css,
  title,
  homeLink,
  fullWidth,
  tall,
  navigation,
  links,
  logo,
  ...otherProps
}) => (
  <HeaderWrapperStyle {...{ css, tall }} id="header">
    <HeaderStyle {...{ fullWidth, ...otherProps }}>
      <HeaderLogoText {...{ tall, title, homeLink, logo }} />
      <HeaderNavigation>{navigation}</HeaderNavigation>
      <HeaderLinks>{links}</HeaderLinks>
    </HeaderStyle>
  </HeaderWrapperStyle>
)

Header.defaultProps = {
  tall: false,
  fullWidth: true,
  logo: AmsterdamLogo,
}

export default Header
