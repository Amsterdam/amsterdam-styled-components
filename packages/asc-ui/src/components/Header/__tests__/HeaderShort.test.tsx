import * as React from 'react'
import Header from '../Header'
import { renderWithTheme } from '../../../utils/withTheme'
import 'jest-styled-components'

describe('Header short', () => {
  it('should render the Header short', () => {
    const component = renderWithTheme(
      <Header
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        search={<div> Search </div>}
        menu={<div> M </div>}
        headerSize="short"
      />,
    )
    expect(component).toMatchSnapshot()
  })
})

describe('Header short with content', () => {
  it('should render the Header short', () => {
    const component = renderWithTheme(
      <Header
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        search={<div> Search </div>}
        menu={<div> M </div>}
        headerSize="short-content"
      />,
    )
    expect(component).toMatchSnapshot()
  })
})

describe('HeaderTitle short', () => {
  it('should render the Header HeaderTitle short', () => {
    const component = renderWithTheme(
      <Header
        title="Data en informatie"
        homeLink="http://data.amsterdam.nl"
        search={<div> Search </div>}
        menu={<div> M </div>}
        headerSize="short-content"
      />,
    )
    expect(component).toMatchSnapshot()
  })
})
