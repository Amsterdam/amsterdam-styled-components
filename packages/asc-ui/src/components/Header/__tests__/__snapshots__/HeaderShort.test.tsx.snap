// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header short should render the Header short 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 50px;
  width: 100%;
  margin: 0 auto;
  background-color: #ffffff;
  max-width: none;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  max-width: 1800px;
  height: 50px;
  width: 100%;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  max-width: none;
}

.c3 {
  display: inline-block;
  margin: 10px 10px 0px 10px;
}

.c3 > svg {
  height: 30px;
  width: 64px;
}

.c2 > h1 {
  height: 100%;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi';
  font-weight: bold;
}

.c2 > h1 > a {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 > h1 > div > span {
  display: inline-block;
}

.c2 > h1 > a > div {
  height: 100%;
  line-height: 33px;
}

.c4 {
  overflow: hidden;
}

@media screen and (min-width:1024.02px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (max-width:768px) {
  .c2 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media screen and (max-width:375px) {
  .c2 > h1 > a > div > span {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c2 > h1 > a > div > span {
    font-size: 16px;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    width: 50px;
  }
}

@media screen and (min-width:768.02px) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media screen and (max-width:1024px) {
  .c5 {
    width: 50px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1>
        <a
          href="http://data.amsterdam.nl"
        >
          <span
            class="c3"
          >
            SVG_MOCK
          </span>
          <div>
            <span>
              Data en informatie
            </span>
          </div>
        </a>
      </h1>
    </div>
    <div
      class="c4"
    >
      <div>
         Search 
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
         M 
      </div>
    </div>
  </div>
</div>
`;

exports[`Header short with content should render the Header short 1`] = `
.c1 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  min-height: 50px;
  width: 100%;
  margin: 0 auto;
  background-color: #ffffff;
  max-width: 1600px;
}

.c0 {
  background-color: #ffffff;
  margin: 0 auto;
  max-width: 1800px;
  height: 50px;
  width: 100%;
  box-shadow: 0 4px rgba(0,0,0,0.04);
  position: fixed;
  top: 0;
  left: 0;
  right: 0;
  max-width: 1800px;
}

.c3 {
  display: inline-block;
  margin: 10px 10px 0px 10px;
}

.c3 > svg {
  height: 30px;
  width: 64px;
}

.c2 > h1 {
  height: 100%;
  margin: 0;
  font-family: 'AvenirNextLTW01-Demi';
  font-weight: bold;
}

.c2 > h1 > a {
  display: -webkit-inline-box;
  display: -webkit-inline-flex;
  display: -ms-inline-flexbox;
  display: inline-flex;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: start;
  align-content: flex-start;
  color: #000;
  -webkit-text-decoration: none;
  text-decoration: none;
}

.c2 > h1 > div > span {
  display: inline-block;
}

.c2 > h1 > a > div {
  height: 100%;
  line-height: 33px;
}

.c4 {
  overflow: hidden;
}

@media screen and (min-width:1024.02px) {
  .c0 {
    display: -webkit-box;
    display: -webkit-flex;
    display: -ms-flexbox;
    display: flex;
  }
}

@media screen and (max-width:768px) {
  .c2 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media screen and (max-width:375px) {
  .c2 > h1 > a > div > span {
    font-size: 14px;
  }
}

@media screen and (min-width:375.02px) {
  .c2 > h1 > a > div > span {
    font-size: 16px;
  }
}

@media screen and (max-width:768px) {
  .c4 {
    width: 50px;
  }
}

@media screen and (min-width:768.02px) {
  .c4 {
    -webkit-box-flex: 1;
    -webkit-flex-grow: 1;
    -ms-flex-positive: 1;
    flex-grow: 1;
  }
}

@media screen and (max-width:1024px) {
  .c5 {
    width: 50px;
  }
}

<div
  class="c0"
>
  <div
    class="c1"
  >
    <div
      class="c2"
    >
      <h1>
        <a
          href="http://data.amsterdam.nl"
        >
          <span
            class="c3"
          >
            SVG_MOCK
          </span>
          <div>
            <span>
              Data en informatie
            </span>
          </div>
        </a>
      </h1>
    </div>
    <div
      class="c4"
    >
      <div>
         Search 
      </div>
    </div>
    <div
      class="c5"
    >
      <div>
         M 
      </div>
    </div>
  </div>
</div>
`;
